onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -249,16 +251,27 @@ - (void)dealloc {
     switch (type) {
       case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
         output_data_byte_size = num_elements * sizeof(float);
+        memcpy(output_tensor.buffer, model_output_data, output_data_byte_size);
         break;
       case ONNX_NAMESPACE::TensorProto_DataType_INT32:
         output_data_byte_size = num_elements * sizeof(int32_t);
+        memcpy(output_tensor.buffer, model_output_data, output_data_byte_size);
+        break;
+      case ONNX_NAMESPACE::TensorProto_DataType_INT64:
+        output_data_byte_size = num_elements * sizeof(int64_t);
+        if (model_output_type == MLMultiArrayDataTypeInt32) {
+          int32_t* model_output_data_prime = static_cast<int32_t*>(model_output_data);
+          int64_t* output_tensor_buffer_prime = static_cast<int64_t*>(output_tensor.buffer);
+          for (size_t i = 0; i < num_elements; i++) {
+            output_tensor_buffer_prime[i] = model_output_data_prime[i];
+          }
+        }
```

Comment:
Return error status if model_output_type != MLMultiArrayDataTypeInt32?
Or add ORT_RETURN_IF_NOT(model_output_type == MLMultiArrayDataTypeInt32, ".....")?

Reply:
it could go before the conditional and replace it
```
ORT_RETURN_IF_NOT(model_output_type == MLMultiArrayDataTypeInt32, ...);
// now we assume model_output_type == MLMultiArrayDataTypeInt32
// ...
```